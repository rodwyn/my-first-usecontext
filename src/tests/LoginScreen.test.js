import { mount } from 'enzyme';
import React from 'react';
import { LoginScreen } from '../LoginScreen';
import { UserContext } from '../UserContext';

describe('Testing <LoginScreen /> component.', () => {
  const setUser = jest.fn();
  const wrapper =  mount(
  <UserContext.Provider value={{
    setUser
  }}>
    <LoginScreen />);
  </UserContext.Provider>);

  test('should render correctly.', () => {
    expect(wrapper).toMatchSnapshot();
  });


  test('should execute setuser function.', () => {
    wrapper.find('button').prop('onClick')();
    expect(setUser).toHaveBeenCalledWith({
      id: '007',
      name: 'Rodwyn Moreno',
      email: 'rodwynmoreno@gmail.com'
    });
  });
});