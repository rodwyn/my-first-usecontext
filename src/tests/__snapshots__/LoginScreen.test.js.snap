// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <LoginScreen /> component. should render correctly. 1`] = `
<div>
  <h1>
    LoginScreen
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
